c/cmake_modules/AdbcDefines.cmake

```
@@ -90,6 +89,8 @@ if(MSVC)
   add_compile_options(/wd4868)
   add_compile_options(/wd4710)
   add_compile_options(/wd4711)
+  # Don't warn about enforcing left-to-right evaluation order for operator[]
+  add_compile_options(/wd4866)
   # Don't warn about padding added after members
   add_compile_options(/wd4820)
```

We may want to sort by warning ID:

```suggestion
  # Don't warn about padding added after members
  add_compile_options(/wd4820)
  # Don't warn about enforcing left-to-right evaluation order for operator[]
  add_compile_options(/wd4866)
```